File: abc/abc121c.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
typedef long long ll;

bool sortbysec(const pair<int, int> &a, const pair<int, int> &b) {
    return a.first < b.first;
}

int main() { 
    int N, M; cin >> N >> M;
    vector<pair<ll, ll>> D;

    for (int i=0; i<N; i++) {
        ll a, b; cin >> a >> b;
        D.push_back(make_pair(a, b));
    }
    sort(D.begin(), D.end(), sortbysec);

    ll cost = 0;
    int ct = 0;
    for (auto v: D) {
        ll rest = M - ct;
        ll buy = min(rest, v.second);
        cost += buy * v.first;
        ct += buy;
    }
    cout << cost << endl;
    
    return 0;
}

File: abc/abc121c.py
def main():
    N, M = map(int, input().split())
    d = []
    for _ in range(N):
        a, b = map(int, input().split())
        d.append((a, b))

    d.sort(key=lambda x: x[0])

    left = M
    cost = 0
    for a, b in d:
        if left < b:
            cost += a * left
            break
        else:
            cost += a * b
            left -= b
    print(cost)


main()
